ggshield/cmd/iac/scan/scan.py

```diff
@@ -97,7 +78,14 @@ def scan_cmd(
         output_handler_cls = IaCTextOutputHandler
     config: Config = ctx.obj["config"]
     output_handler = output_handler_cls(verbose=config.user_config.verbose)
-    return output_handler.process_scan(scan)
+
+    return (
+        output_handler.process_scan(IaCPathScanCollection(scan.id, scan.result))
+        if since is None
+        else output_handler.process_diff_scan(
+            IaCDiffScanCollection(scan.id, scan.result)
+        )
+    )
```

Comment:
@cyrvan-b 
This looks ugly to me. What do you think? Here is the situation:
- There are two result classes, IaCPathScanCollection and IaCDiffScanCollection, for the two scan types. They both inherit an abstract class IaCScanCollection
- There are (currently) two output handlers, IaCTextOutputHandler & IaCJSONOutputHandler, both inheriting an abstract class IaCOutputHandler.

Both types of scan result can use any of the output handler classes.

Reply:
OK I think having two different process methods in the output handlers will be necessary anyway. Adding a `process` method to the ScanCollection classes might work. I'm still open to hearing your opinion on this.

Reply:
We decided not do this this, as it would cause circular dependencies.
Instead, one function will be created in scan.py for each scan result type.